app/models/appointment.rb

```diff
@@ -51,16 +51,26 @@ def self.between(start_date, end_date)
     where("scheduled_date BETWEEN ? and ?", start_date, end_date)
   end
 
-  def self.all_overdue
-    where(status: "scheduled")
-      .where(arel_table[:scheduled_date].lt(Date.current))
-      .where(arel_table[:remind_on].eq(nil).or(arel_table[:remind_on].lteq(Date.current)))
+  def self.passed_unvisited
+    # Scheduled or cancelled appointments whose scheduled date has passed.
+    where.not(appointments: {status: :visited})
+      .where("appointments.scheduled_date < ?", Date.current)
       .joins(:patient)
       .where.not(patients: {status: :dead})
   end
```

Comment:
A new definition `passed_unvisited` is being added. It is a superset of `all_overdue`.

Reply:
Ignoring my longer comment about overall naming, I think we can find something slightly better for this name in the short-term. Nit: `past_unvisited`? `missed` (this incorporates the condition of being in the past in the name itself. You can't miss something in the future)?

Reply:
We originally landed on `past_unvisited`. Wanted to avoid `missed_unvisited` because "Missed visit" is a thing.